src/proportional-bar/ProportionalBar.js

```diff
@@ -0,0 +1,132 @@
+
+const ProportionalBarContainer = styled.figure`
+  height: 100%;
+  margin: 0;
+  position: relative;
+  width: 100%;
+`;
+
+const ProportionalBarChartWrapper = styled.div`
+  height: 100%;
+  position: relative;
+  z-index: ${(props) => props.theme.zIndex.base + 1};
+`;
+
+const ProportionalBarMetadata = styled.figcaption`
+  align-items: baseline;
+  bottom: 0;
+  display: flex;
+  justify-content: space-between;
+  padding-top: 10px;
+  position: absolute;
+  width: 100%;
+  z-index: ${(props) => props.theme.zIndex.base};
+`;
+const ProportionalBarTitle = styled.div`
+  color: ${(props) => props.theme.colors.body};
+  flex: 0 0 auto;
+  font: ${(props) => props.theme.fonts.body};
+  margin-right: 15px;
+`;
+
+const ProportionalBarTooltipText = styled.p`
+  ${(props) => (props.bold ? `font: ${props.theme.fonts.bodyBold};` : "")}
+  line-height: 2;
+  margin: 0;
+  white-space: nowrap;
+`;
+
+// if the legend doesn't wrap it should be roughly this size;
+// the chart height will adjust as necessary
+const INITIAL_METADATA_HEIGHT = 28;
+
+const formatAsPct = format(".0%");
+
+export default function ProportionalBar({ data, title }) {
+  // to both avoid janky chart animations and avoid height overflows
+  // when the legend wraps to multiple lines, we need to measure its height
+  // explicitly to determine the chart height
+  const [metadataHeight, setMetadataHeight] = useState(INITIAL_METADATA_HEIGHT);
+
+  // calculate percentages for display
+  const totalValue = sum(data.map(({ value }) => value));
+  const dataWithPct = data.map((record) => ({
+    ...record,
+    pct: formatAsPct(record.value / totalValue),
+  }));
+
+  const TOOLTIP_PADDING = 3;
+
+  return (
+    <ProportionalBarContainer>
+      <ProportionalBarChartWrapper>
+        <ResponsiveOrdinalFrame
+          data={dataWithPct}
+          // bottom margin leaves room for the title and legend
+          margin={{ top: 0, left: 0, right: 0, bottom: metadataHeight }}
+          oAccessor={() => title}
+          pieceHoverAnnotation
+          projection="horizontal"
+          rAccessor="value"
+          renderKey="label"
+          responsiveHeight
+          responsiveWidth
+          style={(d) => ({ fill: d.color })}
+          tooltipContent={(d) => {
+            return (
+              <Tooltip
+                style={{
+                  // d.y is the vertical center of the hover target; since we know there is only one bar,
+                  // we know it is the vertical center of the entire chart and we can use it
+                  // to push the tooltip below the chart
+                  transform: `translateX(-50%) translateY(${
+                    d.y + TOOLTIP_PADDING
+                  }px)`,
+                }}
+              >
+                <ProportionalBarTooltipText>
+                  {d.label}
+                </ProportionalBarTooltipText>
+                <ProportionalBarTooltipText bold>
+                  {d.value} ({d.pct})
```

Comment:
Are there any charts in the current mocks that would use this proportional bar and where the percentage would be undesired?

Reply:
none that I know of so far ... I don't think that's too likely but it wouldn't be too hard to make it optional if the need arises